### PR TITLE
[hotfix]build a redirect routing for search

### PR DESCRIPTION
**Pull request by chennan47**
<b>Purpose</b>
Resolve the problem that go to any non-existent route prefixed with osf.io/search and you get a 404 page, but there is no search bar at the top anymore. 
image

<b>Changes</b>
Create a new routing for search which process any random string which prefixed by osf.io/search and make them the actual search input and redirect you to the search result page instead of a page 404.
screen shot 2015-02-20 at 1 41 39 pm
will redirect you to page
screen shot 2015-02-20 at 1 42 12 pm

partially resolves https://github.com/CenterForOpenScience/osf.io/issues/1795

### CODE REVIEW

**Comment by fabianvf** [comment]
What is the behavior for the route ```/searchdsfsdgasgdf```?

**Comment by chennan47** [comment]
@fabianvf good catch. Actually that will still lead to a 404 page and have search appear on the first word of index and as a result the search bar will not show up. I will look into that now.

**Comment by chennan47** [comment]
@fabianvf the problem you pointed out is resolved in 3a3dc39

**Comment by sloria** [comment]
I'm not sure that we want to support arbitrary `/search/<query/` routes--keep in mind that we would have to maintain backwards compatibility, and the clashes this would cause if we ever wanted to add additional routes on top of `/search/`, e.g. `/search/public/`.

For now, we just want to make sure that the search bar is available on all 404 pages. @chennan47 Can you make this modification?

File: website/static/js/navbarControl.js [hunk 1]
```diff
@@ -16,9 +16,11 @@ var NavbarViewModel = function() {
     self.showClose = true;
 
     self.onSearchPage = ko.computed(function() {
-        var path = window.location.pathname;
-        var indexOfSearch = path.indexOf('search');
-        return indexOfSearch === 1;
+        if (window.contextVars.search){
```
Comment (sloria) [inline website/static/js/navbarControl.js]: `return window.contextVars.search`

Comment (chennan47) [inline website/static/js/navbarControl.js]: resolved in 44d4d97

**Pull request merged**